### PR TITLE
feat: force activate countly on dev environments

### PR DESCRIPTION
**Pull request by aweiss-dev**
## Description

We want to collect data on our dev environments. 
This PR force activates Data collection on Edge, Dev, Internal, Localhost and Staging environments.

### CODE REVIEW

File: src/script/page/MainContent/panels/preferences/accountPreferences/DataUsageSection.tsx [hunk 1]
```diff
@@ -73,8 +76,9 @@ const DataUsageSection = ({propertiesRepository, brandName, isActivatedAccount}:
               propertiesRepository.savePreference(PROPERTIES_TYPE.PRIVACY.TELEMETRY_SHARING, isChecked);
               setOptionTelemetry(isChecked);
             }}
-            checked={optionTelemetry}
+            checked={isProduction ? optionTelemetry : true}
```
Comment (przemvs) [inline src/script/page/MainContent/panels/preferences/accountPreferences/DataUsageSection.tsx:79]: We want to have always set on true while we not on prod ? 🤔

File: src/script/tracking/Countly.helpers.ts [hunk 2]
```diff
@@ -17,9 +17,17 @@
...
+import {getWebEnvironment} from 'Util/Environment';
+
 import {Config} from '../Config';
 
 export function isCountlyEnabledAtCurrentEnvironment(): boolean {
+  const {isDev, isEdge, isInternal, isLocalhost, isStaging} = getWebEnvironment();
+
+  if (isDev || isEdge || isInternal || isLocalhost || isStaging) {
+    return true;
+  }
```
Comment (przemvs) [inline src/script/tracking/Countly.helpers.ts:29]: Can be simplified like that? :)

```suggestion
  if (!isProduction) {
    return true;
  }
```

File: src/script/page/MainContent/panels/preferences/accountPreferences/DataUsageSection.tsx (hunk 1 above)
Comment (przemvs) [reply src/script/page/MainContent/panels/preferences/accountPreferences/DataUsageSection.tsx:79]: Ahh.. see disabled method.

File: src/script/tracking/EventTrackingRepository.ts [hunk 3]
```diff
@@ -71,6 +72,12 @@ export class EventTrackingRepository {
     this.isErrorReportingActivated = false;
     this.isProductReportingActivated = false;
     amplify.subscribe(WebAppEvents.USER.EVENT_FROM_BACKEND, this.onUserEvent);
+
+    const {isDev, isEdge, isInternal, isLocalhost, isStaging, name} = getWebEnvironment();
+    if (isDev || isEdge || isInternal || isLocalhost || isStaging) {
+      this.forceActivateErrorReporting = true;
```
Comment (przemvs) [inline src/script/tracking/EventTrackingRepository.ts:78]: Can be !isProduction ?

File: src/script/util/Environment.ts [hunk 4]
```diff
@@ -28,6 +28,15 @@ const APP_ENV = {
   VIRTUAL_HOST: 'wire.ms', // The domain "wire.ms" is our virtual host for testing contact uploads
 };
 
+const DEV_ENVIRONMENT_IDENTIFIERS = {
```
Comment (przemvs) [inline src/script/util/Environment.ts:31]: PROD is missing ? 🤔

File: src/script/util/Environment.ts [hunk 5]
```diff
@@ -38,6 +47,44 @@ const isLocalhost = (): boolean => [APP_ENV.LOCALHOST, APP_ENV.VIRTUAL_HOST].inc
...
+    isStaging: false,
+    isDev: false,
+    isInternal: false,
+    isLinked: false,
+    isLocalhost: false,
+    isProduction: false,
+    name: '',
+  };
+
+  if (appBase.includes(DEV_ENVIRONMENT_IDENTIFIERS.LOCAL || isLocalhost())) {
```
Comment (przemvs) [inline src/script/util/Environment.ts]: And maybe we can use for this case swtich/case method ? 🤔 
I think it should be like that :D 

```suggestion
  if (appBase.includes(DEV_ENVIRONMENT_IDENTIFIERS.LOCAL) || isLocalhost()) {
```

File: src/script/tracking/Countly.helpers.ts (hunk 2 above)
Comment (aweiss-dev) [reply src/script/tracking/Countly.helpers.ts:29]: Not the same

File: src/script/tracking/EventTrackingRepository.ts (hunk 3 above)
Comment (aweiss-dev) [reply src/script/tracking/EventTrackingRepository.ts:78]: Not the same

File: src/script/util/Environment.ts (hunk 4 above)
Comment (aweiss-dev) [reply src/script/util/Environment.ts:31]: Doesnt need to be there

File: src/script/util/Environment.ts (hunk 5 above)
Comment (aweiss-dev) [reply src/script/util/Environment.ts]: I like if else more, since we iterate not just over one object

**Review by przemvs — APPROVED** [review]

File: src/script/util/Environment.ts [hunk 6]
```diff
@@ -38,6 +47,44 @@ const isLocalhost = (): boolean => [APP_ENV.LOCALHOST, APP_ENV.VIRTUAL_HOST].inc
...
+  } else if (appBase.includes(DEV_ENVIRONMENT_IDENTIFIERS.LINK)) {
+    environment.isLinked = true;
+    environment.name = 'Linked Environment';
+  } else if (isProduction()) {
+    environment.isProduction = true;
+    environment.name = 'Production Environment';
+  }
+
+  return environment;
+};
```
Comment (e-maad) [inline src/script/util/Environment.ts:88]: ~Wouldn't you mind if we convert these in switch case statements for readability purposes?~

File: src/script/tracking/EventTrackingRepository.ts [hunk 7]
```diff
@@ -251,7 +269,7 @@ export class EventTrackingRepository {
       return;
     }
 
-    if (this.isProductReportingActivated === true) {
+    if (this.isProductReportingActivated === true || this.forceActivateErrorReporting) {
```
Comment (e-maad) [inline src/script/tracking/EventTrackingRepository.ts:272]:
```suggestion
    if (this.isProductReportingActivated || this.forceActivateErrorReporting) {
```

File: src/script/tracking/EventTrackingRepository.ts [hunk 8]
```diff
@@ -268,7 +286,7 @@ export class EventTrackingRepository {
...
```
Comment (e-maad) [inline src/script/tracking/EventTrackingRepository.ts:289]:
```suggestion
    if (this.isProductReportingActivated || this.forceActivateErrorReporting) {
```

File: src/script/tracking/Countly.helpers.ts (hunk 2 above)
Comment (e-maad) [inline src/script/tracking/Countly.helpers.ts:29]:
```suggestion
  const {isProduction, isLinked} = getWebEnvironment();

  if (!isProduction &&  !isLinked) {
    return true;
  }
```
IMO, the opposite conditions are less.

File: src/script/tracking/EventTrackingRepository.ts [hunk 9]
```diff
@@ -71,6 +72,12 @@ export class EventTrackingRepository {
     this.isErrorReportingActivated = false;
     this.isProductReportingActivated = false;
     amplify.subscribe(WebAppEvents.USER.EVENT_FROM_BACKEND, this.onUserEvent);
+
+    const {isDev, isEdge, isInternal, isLocalhost, isStaging, name} = getWebEnvironment();
+    if (isDev || isEdge || isInternal || isLocalhost || isStaging) {
```
Comment (e-maad) [inline src/script/tracking/EventTrackingRepository.ts:77]: Same as above.
I would suggest if you create another variable in `getWebEnvironment()` and reuse it, for example:
```
isExternal = isProduction ||  isLinked;
```

**Review by e-maad — COMMENTED** [review]
Great 🚀 
Added few suggestions.

File: src/script/tracking/Countly.helpers.ts (hunk 2 above)
Comment (aweiss-dev) [reply src/script/tracking/Countly.helpers.ts:29]: yeah but i would like to have them explicit named for readability :)

**Pull request merged**